app/controllers/users/sessions_controller.rb

```diff
@@ -0,0 +1,11 @@
+class Users::SessionsController < Devise::SessionsController
+  def after_sign_in_path_for(resource)
+    return root_path unless resource.developer
```

Comment:
This breaks existing behavior. If you checkout `main`, clicking "Add my profile" then signing in will redirect to editing your profile. Is there a way to call `super` or something and only override the behavior if the resource is a `developer` or something? Or maybe we need to rethink this override?

Reply:
When a user signs in, if they have a developer [profile], they'll be redirected to the edit view. That's the expectation from #50 ?

The 'resource' relates to the the 'user' record, so this line says: 'return the root path unless the user has a developer profile' e.g. any users who have confirmed their account but not yet created a developer [profile] are redirected to the root path. 

Do you think a better approach would be: upon confirming a user account, users should be redirected to create a developer profile i.e. all requests are intercepted to ensure a developer profile exists, if not the user is redirected to the new developer form?

Reply:
Good point! Here's what I think the flows should be:

1. Confirm account -> redirect to creating new developer profile
2. Click "Add your profile" -> sign in with existing with account with developer profile -> redirect to edit your profile
3. Sign in (by manually navigating to `/users/sign_in`) -> redirect to referrer or `root_path`

Reply:
> Confirm account -> redirect to creating new developer profile

What is your expectation if the user does navigates away from the [new developer] form without submitting it?

Would you like to prevent an authenticated user from navigating elsewhere in the application until they have created a profile? If so, overriding the `after_sign_in_path_for` is not the correct place to achieve this, as this method is only catering for a redirect post authentication.

If you do want to enforce developer profile creation (as a kind of next step after confirming the user account), I've implemented this before, so I can always open a PR if that's the preferred approach.

Reply:
> Would you like to prevent an authenticated user from navigating elsewhere in the application until they have created a profile? If so, overriding the after_sign_in_path_for is not the correct place to achieve this, as this method is only catering for a redirect post authentication.

I agree with @davegudge my approach does not work for if we want to disallow a user to visit pages until they have a profile created.

If this is desired behavior, we can have a `before_action :developer!` that will check if the user is developer, if not they will be redirect to create profile page?

Reply:
> we can have a before_action :developer!

Yes, that's what I was thinking. 

With regard to addressing #50, I don't think any redirects are required. The unauthenticated user is clicking 'Add my profile', presented with the sign in screen, and upon success authentication, using devise's standard implementation of 'redirecting to the referrer' should suffice? 

I believe the issue outlined in #50 is that the flash message 'You are already signed in.' is being displayed?

Reply:
> I believe the issue outlined in #50 is that the flash message 'You are already signed in.' is being displayed?

I believe so, yes. So, if we could remove the message I think this PR will also become obsolete. Then removing the message would be fine, and perhaps even desirable?. 
CC @joemasilotti

Reply:
One solution is to blank out the [`devise.failure.already_authenticated`](https://github.com/joemasilotti/railsdevs.io/blob/main/config/locales/devise.en.yml#L10) i18n entry. 

Another i18n entry I tend to blank out is [`devise.sessions.signed_in`](https://github.com/joemasilotti/railsdevs.io/blob/main/config/locales/devise.en.yml#L49) as it's stating the obvious.